src/basho_bench_worker.erl

```diff
@@ -201,8 +210,12 @@ worker_idle_loop(State) ->
     Driver = State#state.driver,
     receive
         {init_driver, Caller} ->
-            %% Spin up the driver implementation
-            case catch(Driver:new(State#state.id)) of
+            %% Spin up the driver implementation, optionally support new approach of passing State
+            DriverNew = case State#state.api_pass_state of 
+                            false -> catch(Driver:new(State#state.id));
+                            _ -> catch(Driver:new(State#state.id, State))
+                        end,
```

Comment:
This should be four space indentation.

One alternative approach to passing `api_pass_state` is to check and see if the driver has a `new/2` function and just call that with the State if it does.